### ISSUE TITLE
Error messages (32)

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/python-jsonpreprocessor#330]
Previously this code

```
{
   "keyP"   : "A",
   ${keyP}[${keyP}] : "newvalue"
}
```

caused

`'Identified dynamic name of key '${keyP}[${keyP}]' that does not exist. But new keys can only be created based on hard code names.'`

With latest changes of the JsonPreprocessor values are returned

`DotDict({'keyP': 'A', 'A': 'newvalue'})`

Both is wrong.

A string cannot be an index of a string.

Expected is error message

`TypeError: 'str' object does not support item assignment`

**Comment by namsonx** [comment]
Hello Holger,

The error message in this case now is:
```
Exception: Could not set parameter '${keyP}[${keyP}]' with value 'newvalue'! Reason: string indices must be integers
```

Thank you,
Son

**Comment by HolQue** [comment]
Retest successful. Issue can be closed.

**Comment by test-fullautomation** [comment]
merged for version 0.13.1
